skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/projectgen/circuitgen/CircuitComponent.kt

```diff
@@ -0,0 +1,408 @@
+        .addAnnotation(IMMUTABLE)
+        .addModifiers(KModifier.SEALED)
+        .addSuperinterface(CIRCUIT_UI_EVENT)
+        .build()
+
+    val stateClass =
+      TypeSpec.classBuilder("State")
+        .addModifiers(KModifier.DATA)
+        .primaryConstructor(
+          FunSpec.constructorBuilder()
+            .addParameter(
+              ParameterSpec.builder(
+                  "eventSink",
+                  LambdaTypeName.get(
+                    receiver = eventInterfaceCN,
+                    returnType = Unit::class.asTypeName(),
+                  ),
+                )
+                .defaultValue("{}")
+                .build()
+            )
+            .build()
+        )
+        .addProperty(
+          PropertySpec.builder(
+              "eventSink",
+              LambdaTypeName.get(receiver = eventInterfaceCN, returnType = Unit::class.asTypeName()),
+            )
+            .initializer("eventSink")
+            .build()
+        )
+        .addSuperinterface(CIRCUIT_UI_STATE)
+        .build()
+    val typeSpec =
+      TypeSpec.classBuilder(screenClassName)
+        .addSuperinterface(SCREEN_INTERFACE)
+        .addAnnotation(PARCELIZE)
+        .addType(stateClass)
+        .addType(eventInterface)
+        .build()
+    return FileSpec.get(packageName, typeSpec)
+  }
+}
+
+class CircuitPresenter(
+  private val assistedInjection: AssistedInjectionConfig,
+  private val additionalCircuitInject: Set<ClassName> = setOf(),
+  private val noUi: Boolean = false,
+) : CircuitComponent {
+  override val fileSuffix = "Presenter"
+
+  override fun generate(packageName: String, className: String): FileSpec {
+    /**
+     * Generate Circuit Presenter class, eg: class ${NAME}Presenter @AssistedInject constructor(
+     *
+     * @Assisted private val screen: ${NAME}Screen,
+     * @Assisted private val navigator: Navigator, ) : Presenter<${NAMEScreen.State> {
+     * @Composable override fun present(): FeatureScreen.State { } }
+     * @AssistedFactory
+     * @CircuitInject( ${NAME}Screen::class, UserScope::class, ) fun interface Factory { fun
+     *   create(screen: ${NAME}Screen, navigator: Navigator): ${NAME}Presenter { } } }
+     */
+    val screenClass = ClassName(packageName, screenClassName(className))
+    val assistedInject = ClassName("dagger.assisted", "AssistedInject")
+    val assisted = ClassName("dagger.assisted", "Assisted")
+    val assistedFactory = ClassName("dagger.assisted", "AssistedFactory")
+    val addAssistedInjection = assistedInjection.screen || assistedInjection.navigator
+
+    val factoryBuilder =
+      TypeSpec.funInterfaceBuilder("Factory").apply {
+        if (addAssistedInjection) {
+          addAnnotation(assistedFactory)
+        }
+        addAnnotation(
+          AnnotationSpec.builder(CIRCUIT_INJECT)
+            .apply {
+              addMember("%T::class", screenClass)
+              additionalCircuitInject.forEach { inject -> addMember("%T::class", inject) }
```

Comment:
CircuitInject accepts exactly two args, I think you got something mixed up here?

Reply:
Thanks for catching that. Updated https://github.com/slackhq/slack-gradle-plugin/pull/799/commits/41e015de2bb22f65c68de702f19c193679d02324